Build failure with OSPD

**Describe the bug**
Sample `samples/subsys/mgmt/osdp` build is failed on all targets due to below error:
```
/home/buildslave/srv/toolchain/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: subsys/mgmt/osdp/libsubsys__mgmt__osdp.a(osdp_common.c.obj): in function `sys_csrand_get':
/home/buildslave/srv/zephyr/master/zephyr/nucleo_f401re/nucleo_f401re/samples/subsys/mgmt/osdp/sample.mgmt.osdp/zephyr/include/generated/syscalls/rand32.h:60: undefined reference to `z_impl_sys_csrand_get'
```

**To Reproduce**
`sanitycheck -p nucleo_f401re -T samples/subsys/mgmt/osdp/`

**Expected behavior**
Should build

**Impact**
Break all CIs

--- a/subsys/mgmt/osdp/Kconfig
+++ b/subsys/mgmt/osdp/Kconfig
@@ -70,12 +70,12 @@ config OSDP_PACKET_TRACE
 
 config OSDP_SC_ENABLED
 	bool "Enable OSDP Secure Channel"
+	depends on CSPRING_ENABLED
 	default y
 	select CRYPTO
 	select CRYPTO_MBEDTLS_SHIM
 	select MBEDTLS
 	select MBEDTLS_CIPHER_CCM_ENABLED
-	select ENTROPY_GENERATOR
 	help
 	  Secure the OSDP communication channel with encryption and mutual
 	  authentication.

--- a/subsys/mgmt/osdp/src/osdp.c
+++ b/subsys/mgmt/osdp/src/osdp.c
@@ -16,11 +16,15 @@
 
 LOG_MODULE_REGISTER(osdp, CONFIG_OSDP_LOG_LEVEL);
 
+#ifdef CONFIG_OSDP_SC_ENABLED
 #ifdef CONFIG_OSDP_MODE_PD
 #define OSDP_KEY_STRING CONFIG_OSDP_PD_SCBK
 #else
 #define OSDP_KEY_STRING CONFIG_OSDP_MASTER_KEY
 #endif
+#else
+#define OSDP_KEY_STRING ""
+#endif	/* CONFIG_OSDP_SC_ENABLED */
 
 struct osdp_device {
 	struct ring_buf rx_buf;

--- a/subsys/mgmt/osdp/src/osdp_pd.c
+++ b/subsys/mgmt/osdp/src/osdp_pd.c
@@ -107,7 +107,7 @@ static struct osdp_pd_cap osdp_pd_cap[] = {
 
 static void pd_decode_command(struct osdp_pd *pd, uint8_t *buf, int len)
 {
-	int i, ret = -1, pos = 0, tmp;
+	int i, ret = -1, pos = 0;
 	struct osdp_cmd *cmd;
 
 	pd->reply_id = 0;
@@ -328,7 +328,10 @@ static void pd_decode_command(struct osdp_pd *pd, uint8_t *buf, int len)
 		ret = 0;
 		break;
 	case CMD_CHLNG:
-		tmp = OSDP_PD_CAP_COMMUNICATION_SECURITY;
+		/* Workaround for error: a label can only be part of a
+		 * statement and a declaration is not a statement */
+		;
+		int tmp = OSDP_PD_CAP_COMMUNICATION_SECURITY;
 		if (pd->cap[tmp].compliance_level == 0) {
 			pd->reply_id = REPLY_NAK;
 			pd->cmd_data[0] = OSDP_PD_NAK_SC_UNSUP;

